### ISSUE TITLE
Compose: deactivate() isn't called

### ISSUE THREAD

**Issue by stsje** [aurelia/templating-resources#138]
It seems `deactivate()` never gets called on your view-model, when loaded through the `<compose>` element.

Since `activate()` is called, you would also expect `deactivate()`, when the view-model is removed.

deactivate would be very convenient when disposing the view-model. My own use case is, I have to dispose subscriptions to eventaggregator.

**Comment by CasiOo** [comment]
It might also be affecting components loaded using the `<content></content>` tag.
Needs further testing, but think I ran into it earlier today. Would have to keep it in mind if implementing this enhancement.

```
<template>
  <foo-with-content-tag>
    <component></component>
  </foo-with-content-tag>
</template>
```

**Comment by devmondo** [comment]
we are facing big problem because of this, without deactivate we have no way to dispose even subscriptions or setTimeOut,etc... usually you do this on `detached` or `deactivate` method, in essence  we have same problem as @stsje has

**Comment by EisenbergEffect** [comment]
Why can't you do this in detached or unbind? Those are the places you would typically do that.

**Comment by devmondo** [comment]
they are not being called!

**Comment by EisenbergEffect** [comment]
If `detached` and `unbind` are not being called then that is a bug. Can you report that as a separate issue and provide a simple repro?

(This issue is about a new feature.)

**Comment by devmondo** [comment]
sure i will try something, thanks for fast response

**Comment by EisenbergEffect** [comment]
@devmondo Is the bug there or not? I don't recall seeing you create an issue.

**Comment by devmondo** [comment]
@EisenbergEffect  sorry for delay.

i was testing, and it turned out to be the Aurelia compiler, when we compile elements, none of them or their compiled children get to call `detached` or `unbind` methods, so the way i do it now is to traverse through all compiled HTML elements and get access to their `viewModel` through `au` property and then call those methods manually on their container root element `unbind()` and it works.  but i dont know if it is the correct approach.

**Comment by EisenbergEffect** [comment]
What compiler are you referring to? If it's that old helper that was passed around some months ago...you definitely shouldn't be using that. That wasn't quite right. In any case, if you use the compiler or the `enhance` api directly, when you have a dynamically created `View` you need to manage the lifecycle yourself. So, if you are creating this from inside a component, then you need to implement all the lifecycle methods and pass through to that dynamic View. I assume you have a ViewSlot as well (or you should) to mange the View, in which case you need to manually manage the ViewSlot's lifecycle instead.

Basically, when you dynamically create Views you need to handle enforcing the lifecycle yourself.

**Comment by EisenbergEffect** [comment]
(For the moment we aren'g

**Comment by EisenbergEffect** [comment]
I'm closing this because instead of using deactivate, in this case, detached or unbind could be used instead. Since we aren't planning to implement the full router lifecycle, it would be strange to implement more methods here.

**Comment by davismj** [comment]
Compose elements are a common tool for encapsulating related code. They are more powerful than custom elements for this purpose simply because of the async activation lifecycle. OPs request to clean up subscriptions is a place for the component lifecycle. However, there are cases where the deactivation lifecycle is more appropriate.
